src/sage/categories/modules_with_basis.py

```diff
@@ -2096,8 +2105,32 @@ def map_coefficients(self, f):
                 sage: a = s([2,1]) + 2*s([3,2])                                         # needs sage.combinat sage.modules
                 sage: a.map_coefficients(lambda x: x * 2)                               # needs sage.combinat sage.modules
                 2*s[2, 1] + 4*s[3, 2]
-            """
-            return self.parent().sum_of_terms( (m, f(c)) for m,c in self )
+
+            We can map into a different base ring::
+
+                sage: # needs sage.combinat
+                sage: e = SymmetricFunctions(QQ).elementary()
+                sage: a = 1/2*(e([2,1]) + e([1,1,1])); a
+                1/2*e[1, 1, 1] + 1/2*e[2, 1]
+                sage: b = a.map_coefficients(lambda c: 2*c, ZZ); b
+                e[1, 1, 1] + e[2, 1]
+                sage: b.parent()
+                Symmetric Functions over Integer Ring in the elementary basis
+                sage: b.map_coefficients(lambda c: 1/2*c, ZZ)
+                Traceback (most recent call last):
+                ...
+                TypeError: no conversion of this rational to integer
+            """
+            R = self.parent()
+            if new_base_ring is not None:
+                B = new_base_ring
+                R = R.change_ring(B)
+            elif isinstance(f, Map):
+                B = f.codomain()
+                R = R.change_ring(B)
+            else:
+                B = self.base_ring()
```

Comment:
```suggestion
            if isinstance(f, Map):
                B = f.codomain()
            else:
                B = self.base_ring()
            if new_base_ring is not None:
                B = new_base_ring
            if B is not self.base_ring():
                R = R.change_ring(B)
```
To be consistent with what (multivariate) polynomials do.